### PR TITLE
ci(e2e): publish phase runtime summary

### PR DESCRIPTION
**Pull request by cjagwani**
## Summary

Follow-up to #7100: scheduled and ordinary manual E2E runs now publish a current-run semantic phase runtime table in the existing GitHub Actions scorecard. The table reuses the artifact audit instead of adding live-log noise, and the resume test no longer reports a bookkeeping-only phase that consistently completed in zero seconds.

## Changes

- download this run's E2E artifacts in the read-only scorecard job and append the existing runtime audit table to the GitHub step summary
- report the slowest observed phase's duration and `passed`, `failed`, or `skipped` outcome
- keep missing or invalid progress evidence diagnostic-only so it cannot hide the main E2E result
- remove `record the completed resume contract` from `onboard-resume`; the final meaningful comparison phase now owns target completion
- protect the workflow boundary, native Node loading path, audit rendering, and phase plan with regression tests
- document the automatic current-run table and the existing multi-run audit command

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [x] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [ ] Docs updated for user-facing behavior changes
- [x] Docs not applicable — justification: no user-facing product behavior changed; the E2E maintainer guides are updated.
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [x] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification: the scorecard remains read-only (`actions: read`, `contents: read`), uses an immutable artifact-action pin, treats download/audit failure as diagnostic-only, and has workflow-boundary regression coverage.
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## Documentation Writer Review

- [x] Documentation writer subagent reviewed the completed implementation
- Result: `docs-updated`
- Evidence: `test/e2e/README.md` and `test/e2e/docs/README.md` document the current-run GitHub Actions scorecard table and its phase-duration and outcome fields.
- Agent: Codex Desktop
- PR: #7388

## DGX Station Hardware Evidence

- [ ] Tested on DGX Station
- Tested commit: not applicable
- Station profile/scenario: not applicable
- Result: not applicable
- Supporting evidence: not applicable

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, or `npm run check:diff` passed when hooks were skipped or unavailable
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — `npm exec vitest -- run --project e2e-support test/e2e/support/test-runtime-audit.test.ts test/e2e/support/e2e-scorecard.test.ts test/e2e/support/e2e-scorecard-coordinator.test.ts test/e2e/support/e2e-operations-workflow-boundary.test.ts` (61 passed); `npm run test:e2e-phases:check` (121 tests across 80 files); `npm run test:projects:check`; `npm run typecheck:cli`.
- [ ] Applicable broad gate passed — `npm test` for broad runtime/test-harness changes; `npm run check` for repo-wide validation/coverage changes — command/result: full `e2e-support` run reached 1,343 passes; five discovery timeouts passed on isolated rerun, and one unrelated lifecycle failure reproduced unchanged in the original checkout.
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed
- [ ] `npm run docs` builds without warnings (doc changes only)
- [ ] Doc pages follow the [style guide](https://github.com/NVIDIA/NemoClaw/blob/main/docs/CONTRIBUTING.md) (doc changes only)
- [ ] New doc pages include SPDX header and frontmatter (new pages only)

---

Signed-off-by: Charan Jagwani <cjagwani@nvidia.com>

## Summary by CodeRabbit

- **New Features**
  - GitHub Actions scorecards now add an **E2E Test Phase Runtime** section, including a semantic phase runtime table and slowest-phase details.
  - Runtime reporting now includes each phase’s **duration and outcome**, with the slowest observed phase’s outcome shown as well.
- **Bug Fixes**
  - If runtime progress artifacts are missing or invalid, scorecards render a **fallback “runtime unavailable”** message instead of failing.
- **Documentation**
  - Updated E2E documentation to reflect the expanded runtime tables and enriched scorecard summary behavior for scheduled/manual runs.
- **Tests**
  - Expanded E2E runtime and workflow boundary validations for the new section, artifact handling, and outcome reporting.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -5106,6 +5118,19 @@ jobs:
...
+              runtimeSummaryMarkdown = runtimeAudit.formatRuntimeAuditSummary(runtimeRows);
+            } catch {
+              core.warning('E2E test phase runtime summary unavailable: invalid progress artifact');
+              runtimeSummaryMarkdown = [
+                '## E2E Test Phase Runtime',
+                '',
+                'The summary is unavailable because a `test-progress.json` artifact was invalid.',
+                '',
+              ].join('\n');
+            }
```
[the remark this answers is not in the document]
Comment (cjagwani) [reply .github/workflows/e2e.yaml:5185]: Fixed in b35b68fa3. The runtime audit now runs before `buildTraceTimingResult()`, and the workflow execution test asserts that call order.

File: test/e2e/support/e2e-operations-workflow-boundary.test.ts [hunk 2]
```diff
@@ -64,6 +64,21 @@ describe("E2E operations workflow boundary", () => {
     );
   });
 
+  it("pins the scorecard's current-run progress artifact download", () => {
+    const workflow = readE2eOperationsWorkflow();
+    const download = workflow.jobs.scorecard.steps!.find(
+      (step) => step.name === "Download E2E progress artifacts",
+    )!;
+    download.uses = "actions/download-artifact@0000000000000000000000000000000000000000";
+    download.with!.pattern = "*";
```
[the remark this answers is not in the document]
Comment (cjagwani) [reply test/e2e/support/e2e-operations-workflow-boundary.test.ts:84]: Fixed in b35b68fa3. The action pin and artifact pattern now have separate mutation tests, so each boundary is exercised independently.

**Review by prekshivyas — APPROVED** [review]
Reviewed exact head 25a1775ec85d1e8d5749dfe91686572ad5f4a215 against base 1d418f55b6ea3f2710be2b22785ebf6375d0852f. Security PASS across all nine review categories. The scorecard remains read-only, actions are immutably pinned, artifact traversal skips links, and missing or invalid progress evidence stays diagnostic-only without hiding the E2E result. Focused validation: 61 tests passed. No actionable findings. Exact-SHA credentialed E2E still requires maintainer authorization.

**Comment by cv** [comment]
Addressed PRA-1 in signed commit 867f6735f: the runtime-audit support test now supplies otherwise valid summaries with a missing phase outcome and an unknown phase outcome, and confirms both are rejected. Fresh GitHub checks are running on the new head.
